dotCMS/src/main/java/com/dotcms/publisher/business/PublisherQueueJob.java

```diff
@@ -308,61 +292,107 @@ private void updateAuditStatus() throws DotPublisherException, DotDataException
 	@NotNull
 	private void updateBundleStatus(final PublishAuditStatus bundleAudit,
 									final Map<String, Map<String, EndpointDetail>> endpointTrackingMap,
-									final GroupPushStats groupPushStats) throws DotDataException, DotPublisherException {
+									final GroupPushStats groupPushStats, final List<Map<String, Object>> bundlesInQueue) throws DotDataException, DotPublisherException {
 		Status bundleStatus;
-		PublishAuditHistory localHistory = bundleAudit.getStatusPojo();
-
+		final PublishAuditHistory localHistory = bundleAudit.getStatusPojo();
+		final String auditedBundleId = bundleAudit.getBundleId();
 		if ( localHistory.getNumTries() >= MAX_NUM_TRIES && (groupPushStats.getCountGroupFailed() > 0
                 || groupPushStats.getCountGroupPublishing() > 0) ) {
             // If bundle cannot be installed after [MAX_NUM_TRIES] tries
             // and some groups could not be published
-            List<Environment> environments = APILocator.getEnvironmentAPI().findEnvironmentsByBundleId(bundleAudit.getBundleId());
+            List<Environment> environments = APILocator.getEnvironmentAPI().findEnvironmentsByBundleId(auditedBundleId);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'environments' could be declared final](https://app.codacy.com/app/dotCMS/core/pullRequest?prid=4521280)